Sources/SwiftOperatorPrecedence/OperatorPrecedence.swift

```diff
@@ -0,0 +1,105 @@
+//===------------------ OperatorPrecedence.swift --------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftSyntax
+
+/// Maintains information about operators and their relative precedence,
+/// providing the core operations for "folding" sequence expression syntax into
+/// a structured expression syntax tree.
+public struct OperatorPrecedence {
+  var precedenceGraph: PrecedenceGraph = .init()
+  var operators: [OperatorName : Operator] = [:]
+
+  public init() { }
+
+  /// Record the operator, if it matters.
+  /// FIXME: Terrible API used only for tests
+  public mutating func record(
+    _ op: Operator,
+    errorHandler: OperatorPrecedenceErrorHandler = { throw $0 }
+  ) rethrows {
+    // FIXME: Could do operator-already-exists checking for prefix/postfix
+    // operators as well, since we parse them.
+    if op.kind != .infix { return }
```

Comment:
I was thinking that for prefix/postfix operators we could still use all of the same logic for (e.g.) diagnosing conflicting operators and such. Perhaps that means the whole thing is mis-named, and we should call it the "operator table" rather than "operator precedence".

Reply:
I agree, if this was named `OperatorTable`, using it to diagnose conflicting operators would make sense 👍